jcli/src/jcli_lib/vote/tally/merge_results.rs

```diff
@@ -0,0 +1,344 @@
+                    },
+                )
+            })
+            .collect();
+
+        // indexes of the proposals that have the same external proposal set as vp1
+        let mut matches = voteplans
+            .iter()
+            .enumerate()
+            .filter_map(|(i, vp2)| {
+                if vp2.proposals.len() == p1.len()
+                    && vp2
+                        .proposals
+                        .iter()
+                        .all(|p| p1.get(&p.proposal_id).is_some())
+                {
+                    Some(i)
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>();
+
+        for i in &matches {
+            let vp2 = &voteplans[*i];
+            ids.insert(vp2.id);
+
+            let p2: BTreeMap<Hash, &VoteProposalStatus> = vp2
+                .proposals
+                .iter()
+                .map(|vps| (vps.proposal_id, vps))
+                .collect();
+
+            for (mut ps1, ps2) in p1.values_mut().zip(p2.into_iter().map(|(_, v)| v)) {
+                ps1.votes_cast += ps2.votes_cast;
+                ps1.tally = match (&ps1.tally, &ps2.tally) {
+                    (Tally::Public { result: result1 }, Tally::Public { result: result2 }) => {
+                        Tally::Public {
+                            result: result1.merge(result2),
+                        }
+                    }
+                    (
+                        Tally::Private {
+                            state: PrivateTallyState::Decrypted { result: result1 },
+                        },
+                        Tally::Private {
+                            state: PrivateTallyState::Decrypted { result: result2 },
+                        },
+                    ) => Tally::Private {
+                        state: PrivateTallyState::Decrypted {
+                            result: result1.merge(result2),
+                        },
+                    },
+                    (Tally::Public { result: _ }, Tally::Private { state: _ })
+                    | (Tally::Private { state: _ }, Tally::Public { result: _ }) => {
+                        return Err(Error::PrivacyMismatch);
+                    }
+                    (Tally::Private { state: _ }, Tally::Private { state: _ }) => {
+                        return Err(Error::VotePlanEncrypted)
+                    }
+                };
+            }
+        }
+
+        // already merged proposals shouldn't be considered again
+        //
+        // reverse is necessary because the last index can be in the list
+        matches.reverse();
+        for i in matches {
+            voteplans.swap_remove(i);
+        }
+
+        // FIXME: I'm not actually sure if this should be the correct behavior.
+        //
+        // On one hand, a voteplan without matches can't be merged.
+        //
+        // On the other, it can be considered as is trivially merged.
```

Comment:
Given the current use case, I would at least emit a warning (if not an error) if a voteplan can't be merged